torch/csrc/jit/passes/quantization.cpp

```diff
@@ -886,5 +886,131 @@ graph(%self, %scale, %zero_point, %dtype):
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
+      if (n->kind() == prim::CallMethod) {
+        auto module_instance = n->inputs()[0];
+        auto module_method_name = n->s(attr::name);
+        script::Module callee_module = module;
+        if (module_instance->node()->kind() == prim::GetAttr) {
+          auto child_module_name = module_instance->node()->s(attr::name);
+          auto child_module = module.find_module(child_module_name);
+          TORCH_INTERNAL_ASSERT(
+              child_module,
+              "Child module " + child_module_name + " does not exist");
+          callee_module = child_module.value();
+        } else {
+          TORCH_INTERNAL_ASSERT(
+              module_instance == graph->inputs()[0],
+              "We only support call method either on %self"
+              "or child instance in insert_observers_pass right now");
+        }
+        FoldPrepackedWeightIntoModule(callee_module, method_name, wrapper_module);
+      }
+      for (Block* subblock : n->blocks()) {
+        blocks_to_visit.push(subblock);
+      }
+    }
+  }
+
+  auto patterns = {linear_with_quant};
+  for (const auto& pattern: patterns) {
+    Graph pattern_graph;
+    std::unordered_map<std::string, Value*> vmap;
+    script::parseIR(pattern, &pattern_graph, vmap);
+    const auto& matches = findPatternMatches(pattern_graph, *graph);
+    TORCH_CHECK(
+        matches.size() <= 1, "We only support at most one match right now");
+    for (const auto& match : matches) {
+      const auto& match_vmap = match.values_map;
+      auto w_scale = toIValue(match_vmap.at(vmap.at("w_scale"))).value().toDouble();
+    auto w_zero_point =
+      toIValue(match_vmap.at(vmap.at("w_zero_point"))).value().toInt();
+    auto w_dtype =
+        toIValue(match_vmap.at(vmap.at("w_dtype"))).value().toScalarType();
+      auto w = module.get_parameter("weight").variable_data();
+      auto w_quant = at::quantize_per_tensor(w, w_scale, w_zero_point, w_dtype);
+      auto b = module.get_parameter("bias").variable_data();
+      auto wrapper = wrapper_module.clone();
+      auto set_weight_bias = wrapper.get_method("set_weight_bias");
+      set_weight_bias(std::vector<IValue>{IValue(w_quant), IValue(b)});
+      // TODO: we need to make sure this name is unique
+      module.register_module(
+          "_packed_linear_weight_bias",
+          wrapper
+      );
+
+      // Replace GetAttr on %self with PackedParams module
+      auto w_val = match_vmap.at(vmap.at("w"));
+      auto b_val = match_vmap.at(vmap.at("b"));
+      auto w_quant_val = match_vmap.at(vmap.at("w_quant"));
+
+      // m = self._packed_linear_weight_bias
+      Node* packed_params_module = graph->create(prim::GetAttr);
+      packed_params_module->addInput(graph->inputs()[0]);
+      packed_params_module->s_(attr::name, "_packed_linear_weight_bias");
+      packed_params_module->output()->setDebugName("m");
```

Comment:
Usually when we set a debug name, we get a name of an existing value in the graph and append it with some suffix. Setting names like "m" will screw up the existing names "m" which are pretty likely to be there from the original (user's) code. It's all only debug names, so everything would work correctly anyways, but I don't think we should destroy the existing names if we can avoid that. At least we should prepend the names of generated values with underscores or something to reduce chances of collision with real names.